ashley/src/com/badlogic/ashley/core/Entity.java

```diff
@@ -62,11 +62,15 @@ public Entity () {
 	 * @return The Entity for easy chaining
 	 */
 	public Entity add (Component component) {
-		if (componentOperationHandler != null) {
-			componentOperationHandler.add(this, component);
-		} else {
-			addInternal(component);
+		if (addInternal(component)) {
```

Comment:
Naturally, only schedule operations or notify listeners if the component was actually added/removed.